pkgs/applications/networking/instant-messengers/signalbackup-tools/default.nix

```diff
@@ -1,30 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, openssl, sqlite }:
+{ lib, stdenv, fetchFromGitHub, fetchpatch, openssl, sqlite }:
 
 stdenv.mkDerivation rec {
   pname = "signalbackup-tools";
-  version = "20220711";
+  version = "20220810";
 
   src = fetchFromGitHub {
     owner = "bepaald";
     repo = pname;
     rev = version;
-    sha256 = "sha256-dKU8oTQ6ECwycDN3k7NY/pKpNWH16ceJIFDnRNEA90c=";
+    sha256 = "sha256-z/RAvNUss9rNuBQvxjJQl66ZMrlxvmS9at8L/vSG0XU=";
   };
 
-  buildInputs = [ openssl sqlite ];
-  buildFlags = [
-    "-Wall"
-    "-Wextra"
-    "-Wshadow"
-    "-Wold-style-cast"
-    "-Woverloaded-virtual"
-    "-pedantic"
-    "-std=c++2a"
-    "-O3"
-    "-march=native"
+  patches = [
+    (fetchpatch {
+      name = "fix-platform-checks.patch";
+      url = "https://github.com/bepaald/signalbackup-tools/compare/20220810..a81baf25b6ba63da7d30d9a239e5b4bbc8d1ab4f.patch";
+      sha256 = "sha256-i7fuPBil8zB+V3wHHdcbmP79OZoTfG2ZpXPQ3m7X06c=";
+    })
   ];
-  buildPhase = ''
-    $CXX $buildFlags */*.cc *.cc -lcrypto -lsqlite3 -o signalbackup-tools
+
+  buildInputs = [ openssl sqlite ];
+  buildPhase = lib.optionalString stdenv.isDarwin ''
+    export CXXFLAGS="-Wall -Wextra -Wshadow -Wold-style-cast -Woverloaded-virtual -pedantic -O3"
+    export LDFLAGS="-Wall -Wextra -O3"
```

Comment:
Why are these required on Darwin?

Reply:
In [`BUILDSCRIPT_MULTIPROC.bash44`](https://github.com/bepaald/signalbackup-tools/blob/a81baf25b6ba63da7d30d9a239e5b4bbc8d1ab4f/BUILDSCRIPT_MULTIPROC.bash44) these variables are defined as follows:

```bash
CXXFLAGS="${CXXFLAGS:--Wall -Wextra -Wshadow -Wold-style-cast -Woverloaded-virtual -pedantic -O3 -flto}"
LDFLAGS="${LDFLAGS:--Wall -Wextra -Wl,-z,now -Wl,--as-needed -O3 -s -flto=auto}"
```

Link time optimization currently doesn't work on for Darwin in `nixpkgs`, so the `-flto` flag in `CXXFLAGS` leads to linker errors. The linker on Darwin doesn't support the `-z` or `--as-needed` flags, and the `-s` is obsolete, so those all need to be removed from `LDFLAGS`.